File: pms-backend/src/main/java/com/pms/service/SprintService.java
package com.pms.service;

import com.pms.dto.request.SprintRequest;
import com.pms.dto.response.SprintResponse;

import java.util.List;

public interface SprintService {
    SprintResponse create(SprintRequest request);
    SprintResponse start(Long id);
    SprintResponse complete(Long id);
    List<SprintResponse> getByProject(Long projectId);
    SprintResponse getById(Long id);
}

File: pms-frontend/src/services/sprintService.js
import api from './api';

const sprintService = {
  getByProject: (projectId) => api.get(`/sprints/project/${projectId}`).then((r) => r.data.data),
  getById: (id) => api.get(`/sprints/${id}`).then((r) => r.data.data),
  create: (payload) => api.post('/sprints', payload).then((r) => r.data.data),
  start: (id) => api.patch(`/sprints/${id}/start`).then((r) => r.data.data),
  complete: (id) => api.patch(`/sprints/${id}/complete`).then((r) => r.data.data),
};

export default sprintService;
